src/SonarQube.Client.Tests/Api/SonarQubeService_GetNotificationEventsAsync.cs

```diff
@@ -0,0 +1,133 @@
+      ""project"": ""my_project"",
+      ""date"":""2017-12-20T17:07:34+0100""
+    }
+  ]
+}");
+
+            var result = await service.GetNotificationEventsAsync("my_project", now, CancellationToken.None);
+
+            messageHandler.VerifyAll();
+
+            result.Should().HaveCount(2);
+            result.Select(x => x.Category).Should().BeEquivalentTo(new[] { "QUALITY_GATE", "NEW_ISSUES" });
+            result.Select(x => x.Date).Should().BeEquivalentTo(
+                new[]
+                {
+                    DateTimeOffset.Parse("2017-11-15T17:07:34+0100"),
+                    DateTimeOffset.Parse("2017-12-20T17:07:34+0100")
+                });
+            result.Select(x => x.Link).Should().BeEquivalentTo(
+                new[]
+                {
+                    new Uri("https://sonarcloud.io/dashboard?id=my_project"),
+                    new Uri("https://sonarcloud.io/project/issues?id=my_project&createdAfter=2017-03-01T00%3A00%3A00%2B0100&assignees=me%40github&resolved=false")
+                });
+            result.Select(x => x.Message).Should().BeEquivalentTo(
+                new[]
+                {
+                    "Quality Gate of project 'My Project' is now Red (was Orange)",
+                    "You have 15 new issues on project 'My Project'"
+                });
+        }
+
+        [TestMethod]
+        public async Task GetNotifications_NotFound()
+        {
+            var now = new DateTimeOffset(2017, 10, 19, 13, 0, 0, TimeSpan.FromHours(2));
+
+            await ConnectToSonarQube("6.6.0.0");
+
+            SetupRequest("api/developers/search_events?projects=my_project&from=2017-10-19T13:00:00%2b0200", "",
+                HttpStatusCode.NotFound);
+
+            var result = await service.GetNotificationEventsAsync("my_project", now, CancellationToken.None);
+
+            result.Should().BeNull();
+
+            messageHandler.VerifyAll();
+        }
+
+        [TestMethod]
+        public async Task GetNotifications_OtherError()
+        {
+            var now = new DateTimeOffset(2017, 10, 19, 13, 0, 0, TimeSpan.FromHours(2));
+
+            await ConnectToSonarQube("6.6.0.0");
+
+            SetupRequest("api/developers/search_events?projects=my_project&from=2017-10-19T13:00:00%2b0200", "",
+                HttpStatusCode.InternalServerError);
+
+            var result = await service.GetNotificationEventsAsync("my_project", now, CancellationToken.None);
+
+            result.Should().BeEmpty();
+
+            messageHandler.VerifyAll();
+        }
+
+        [TestMethod]
+        public async Task GetNotifications_Old_SonarQube()
+        {
+            await ConnectToSonarQube("6.5.0.0");
+
+            Func<Task> action = async () =>
+                await service.GetNotificationEventsAsync("my_project", new DateTimeOffset(), CancellationToken.None);
+
+            action.Should().ThrowExactly<InvalidOperationException>();
```

Comment:
Don't we have any expected message for this issue?

Reply:
I'd say leave the test as it is.